build_tools/benchmarks/common/benchmark_presentation.py

```diff
@@ -35,6 +39,103 @@ class AggregateBenchmarkLatency:
   base_mean_time: Optional[int] = None
 
 
+@dataclass
+class CompilationMetrics:
+  """An object for describing the summary of statistics and the reference."""
+  compilation_info: CompilationInfo
+  compilation_time: int
+  total_dispatch_component_size: int
+  base_compilation_time: Optional[int] = None
+  base_total_dispatch_component_size: Optional[int] = None
+
+
+T = TypeVar("T")
+
+
+class MetricsToTableMapper(ABC, Generic[T]):
+  """Abstract class to help map benchmark metrics to table."""
```

Comment:
I think this could use a little more explanation/documentation. It took me a bit to grok what was going on.